drivers/cifutils.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/python
+#
+# Copyright (C) Citrix Systems Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation; version 2.1 only.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# ISOSR: remote iso storage repository
+
+import util
+import xs_errors
+
+class CIFSException(Exception):
+    def __init__(self, errstr):
+        self.errstr = errstr
+
+def containsPassword(dconf):
+    return (('cifspassword' in dconf) or ('cifspassword_secret' in dconf))
+
+def containsCredentials(dconf):
+    return ((('username' in dconf)) and (containsPassword(dconf)))
+
+def splitDomainAndUsername(uname):
+    
+        username = None
+        domain = None
+        dom_username = uname.split('\\')
+
+        if len(dom_username) == 1:
+            domain = None
+            username = dom_username[0]
+        elif len(dom_username) == 2:
+            domain = dom_username[0]
+            username = dom_username[1]
+        else:
+            raise CIFSException("A maximum of 2 tokens are expected "
+                                   "(<domain>\<username>). {} were given."
+                                   .format(len(dom_username)))
+        return username, domain
+
+def getCIFCredentials(dconf, session, username_overide=None):
```

Comment:
What is `username_override`? Doesn't seem to be used anywhere?

Reply:
This was part of my general misunderstanding, will fix